setup.py

```diff
@@ -14,7 +14,6 @@
     "geopandas>=0.10",
     "dask>=2021.06.0",
     "distributed>=2021.06.0",
-    "pygeos",
```

Comment:
With this, you can potentially install dask-gepandas with shapely 1.8 and no pygeos. I know that we cannot do shapely2 or pygeos in requirements but we shouldn't allow pip to resolve environment that is not sufficient to run it. 

That said, I don't know how to do that given the flexibility we have now so it may in the end be the best solution to keep it as you suggest and let people figure out potential issues from docs. 

Btw, how do you want to set requirements in conda-feedstock? The same and assume it will resolve with shapely 2?

Reply:
We could still do a run-time check, and error (or warn) if you have only shapely<2 without pygeos?

> Btw, how do you want to set requirements in conda-feedstock? The same and assume it will resolve with shapely 2?

I assume so, we also removed "pygeos" from the pyogrio feedstock (although for pyogrio, it is less critical to have pygeos or shapely>=2)

Reply:
> We could still do a run-time check, and error (or warn) if you have only shapely<2 without pygeos

That may be wise. Otherwise we may run into some weird behaviour and possibly unclear error messages.

Reply:
Ah, yes, that's true, I forgot that right now you actually get an error in `calculate_spatial_partitions` in such a case. While is not essential functionality (you can use dask-geopandas without ever calling this), we can provide a better error messaging (and since pygeos always has been required in practice by importing it top-level, it's not actually a big change to check for this in `__init__.py`)

Reply:
OK, added a more informative error message. 

I don't think it is worth adding an extra CI build to cover this, but tested locally in an env without pygeos and with older shapely:

```
In [1]: import dask_geopandas
...
ImportError: Running dask-geopandas requires either Shapely >= 2.0 or PyGEOS to be installed. However, PyGEOS is not installed and Shapely is only at version 1.8.0
```